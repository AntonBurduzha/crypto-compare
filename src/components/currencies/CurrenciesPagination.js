import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { bindActionCreators } from'redux';
import * as currenciesActions from '../../actions';
import { PAGE_SIZE } from '../../constants';
import { Pagination } from 'antd';

class CurrenciesPagination extends Component {
  onChangePage = pageNumber => this.props.getNextPage(pageNumber);

  render() {
    const { fullList, page, filteredList, searchedKey } = this.props.currencies;
    const list = searchedKey ? filteredList : fullList;
    return (
      <Fragment>
        { list.length &&
            <Pagination
              total={list.length}
              showTotal={(total, range) => {
                return `${range[0]}-${range[1]} of ${total} items`}
              }
              pageSize={PAGE_SIZE}
              defaultCurrent={1}
              current={page}
              onChange={this.onChangePage}
            />
          }
      </Fragment>
    );
  }
}

function mapStateToProps(state) {
  return { currencies: state.app.currencies };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators({ ...currenciesActions }, dispatch);
}

export default connect(mapStateToProps, mapDispatchToProps)(CurrenciesPagination);
