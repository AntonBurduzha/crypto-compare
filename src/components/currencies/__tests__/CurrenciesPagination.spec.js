import React from 'react';
import { Pagination } from 'antd';
import { shallow } from 'enzyme';
import { CurrenciesPagination, mapStateToProps, mapDispatchToProps } from '../CurrenciesPagination';

describe('CurrenciesPagination component', () => {
  const spy = jest.fn();
  let wrap;
  beforeEach(() => {
    const props = {
      searchedKey: '',
      page: 1,
      fullList: [1, 2, 3],
      filteredList: [1, 2],
      getNextPage: spy,
    };
    wrap = shallow(<CurrenciesPagination {...props}/>);
  });

  it('renders without crashing', () => {
    expect(wrap).toBeDefined();
  });

  it('has Pagination if has searched key', () => {
    expect(wrap.find(Pagination)).toHaveLength(1);
    expect(wrap.containsMatchingElement(<Pagination/>)).toEqual(true);
  });

  it('has Pagination if has no searched key', () => {
    wrap.setProps({ filteredList: [1, 2], searchedKey: 'kek' });
    expect(wrap.find(Pagination)).toHaveLength(1);
    expect(wrap.containsMatchingElement(<Pagination/>)).toEqual(true);
  });

  it('has no Pagination ', () => {
    wrap.setProps({ fullList: [] });
    expect(wrap.find(Pagination)).toHaveLength(0);
    expect(wrap.containsMatchingElement(<Pagination/>)).toEqual(false);
  });

  it('call getNextPage action after clicking on any Pagination btn', () => {
    wrap.find(Pagination).props().onChange(2);
    expect(spy).toHaveBeenCalled();
    expect(spy).toHaveBeenCalledTimes(1);
    expect(spy).toHaveBeenCalledWith(2)
  });

  it('has redux context', () => {
    const initState = {
      app: { currencies: { searchedKey: '', page: 1, fullList: [1, 2, 3], filteredList: [1, 2] } }
    };
    const dispatch = jest.fn();

    expect(mapStateToProps(initState).searchedKey).toEqual('');
    expect(mapStateToProps(initState).page).toEqual(1);
    expect(mapStateToProps(initState).fullList).toEqual([1, 2, 3]);
    expect(mapStateToProps(initState).filteredList).toEqual([1, 2]);
    expect(mapDispatchToProps(dispatch).getNextPage).toBeDefined();
  });

  it('matches snapshot', () => {
    expect(wrap).toMatchSnapshot();
  });
});
