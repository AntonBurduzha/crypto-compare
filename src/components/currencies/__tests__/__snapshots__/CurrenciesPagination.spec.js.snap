// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrenciesPagination component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrenciesPagination
    filteredList={
        Array [
            1,
            2,
          ]
    }
    fullList={
        Array [
            1,
            2,
            3,
          ]
    }
    getNextPage={[Function]}
    page={1}
    searchedKey=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <Pagination
        current={1}
        defaultCurrent={1}
        onChange={[Function]}
        pageSize={32}
        prefixCls="ant-pagination"
        selectPrefixCls="ant-select"
        showTotal={[Function]}
        total={3}
/>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "current": 1,
        "defaultCurrent": 1,
        "onChange": [Function],
        "pageSize": 32,
        "prefixCls": "ant-pagination",
        "selectPrefixCls": "ant-select",
        "showTotal": [Function],
        "total": 3,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <Pagination
          current={1}
          defaultCurrent={1}
          onChange={[Function]}
          pageSize={32}
          prefixCls="ant-pagination"
          selectPrefixCls="ant-select"
          showTotal={[Function]}
          total={3}
/>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "current": 1,
          "defaultCurrent": 1,
          "onChange": [Function],
          "pageSize": 32,
          "prefixCls": "ant-pagination",
          "selectPrefixCls": "ant-select",
          "showTotal": [Function],
          "total": 3,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
